### PR TITLE
ci: demo builds compile all GA libraries

### PR DESCRIPTION
**Pull request by coryan**
I want to avoid any embarrassment if the libraries fail to build for whatever reason.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10634)

### CODE REVIEW

File: google/cloud/channel/CMakeLists.txt [hunk 1]
```diff
@@ -14,12 +14,18 @@
 # limitations under the License.
 # ~~~
 
-# TODO(8125): Re-enable the macos build.
+# TODO(#8125): Re-enable the on macOS and GCC < 8.0.
```
Comment (devbww) [inline google/cloud/channel/CMakeLists.txt:17]: s/the/this/

File: google/cloud/channel/CMakeLists.txt [hunk 2]
```diff
@@ -14,12 +14,18 @@
...
 if (APPLE)
     message(WARNING "Cannot build this google/cloud/channel "
```
Comment (devbww) [inline google/cloud/channel/CMakeLists.txt:19]: s/ this//

File: google/cloud/channel/CMakeLists.txt [hunk 3]
```diff
@@ -14,12 +14,18 @@
...
                     "on Apple platforms until #8125 is fixed.")
     return()
 endif ()
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION
+                                                  VERSION_LESS 8.0)
+    message(WARNING "Cannot build this google/cloud/channel "
```
Comment (devbww) [inline google/cloud/channel/CMakeLists.txt:25]: s/ this//

File: google/cloud/channel/CMakeLists.txt [hunk 4]
```diff
@@ -14,12 +14,18 @@
...
+                    "on with GCC < 8.0 until #8125 is fixed.")
```
Comment (devbww) [inline google/cloud/channel/CMakeLists.txt:26]: s/on //

**Review by devbww — APPROVED** [review]

File: google/cloud/channel/CMakeLists.txt (hunk 1 above)
Comment (coryan) [reply google/cloud/channel/CMakeLists.txt]: Fixed.

File: google/cloud/channel/CMakeLists.txt (hunk 2 above)
Comment (coryan) [reply google/cloud/channel/CMakeLists.txt]: Fixed

File: google/cloud/channel/CMakeLists.txt (hunk 3 above)
Comment (coryan) [reply google/cloud/channel/CMakeLists.txt]: Fixed

File: google/cloud/channel/CMakeLists.txt (hunk 4 above)
Comment (coryan) [reply google/cloud/channel/CMakeLists.txt]: Fixed

**Pull request merged**